CRM/Core/Permission.php

```diff
@@ -1167,6 +1167,8 @@ public static function getEntityActionPermissions() {
     ];
     $permissions['contribution_recur'] = $permissions['payment'];
 
+    $permissions['entity_financial_trxn'] = $permissions['payment'];
```

Comment:
Hmm When we have a payment api (which we don't yet in v4) we will be permitting create.payment fairly widely - it's the financial_trxn we will limit ability to do a create on

Reply:
yeah, and payment API would be one of those abstract entity which got its own set of CRUD and in terms of permission it has to rely on one of the financial entity (here FinancialTrxn). Recently I am working on another abstract entity - Attachment (PR https://github.com/civicrm/civicrm-core/pull/20494) which could be a blueprint for implementing (financial) abstract entities like Payment, Order etc.

Reply:
@monishdeb yeah so if we add a Payment.create api it will probably have quite open permissions (make online contributions, access CiviContribute) whereas the others will be quite narrow - does that align with what you are doing here?

Reply:
@colemanw we should ideally wrap this up pre fork too - but I'm struggling to get my head around what is right. This seems to be INCREASING who can write directly to financial entities by giving the the same permissions as 'payment' ?

Reply:
@monishdeb on re-reading - the ONLY line in this PR I have an issue with is 

```
 $permissions['entity_financial_trxn'] = $permissions['payment'];
```

as it is the only line that is loosening rather than tightening - do tests fail if we remove it?

Reply:
nope, tests (on api3 and api4) won't be affected but APIv4 class need to updated to 
```patch
class EntityFinancialTrxn extends Generic\DAOEntity {  
use Generic\Traits\EntityBridge;
...
public static function permissions() {    
- $permissions = \CRM_Core_Permission::getEntityActionPermissions()['entity_financial_trxn'] ?? []; 
+ $permissions = \CRM_Core_Permission::getEntityActionPermissions()['default'] ?? [];     
...
```